File: 1-easy/problems/3127-make-a-square-with-the-same-color/solution.ts
function canMakeSquare(grid: any): boolean | number | string | any {
  """
  :type grid: List[List[str]]
  :rtype: bool
  """
  N, W = 3, 2
  return any(max(collections.Counter(grid[i+h][j+w] for h in Array.from({length: W}, (_, i) => i) for w in Array.from({length: W}, (_, i) => i)).itervalues()) >= W**2-1;
             for i in Array.from({length: N-W+1}, (_, i) => i) for j in Array.from({length: N-W+1}, (_, i) => i))
}

export { canMakeSquare };

File: 1-easy/problems/3127-make-a-square-with-the-same-color/solution.py
from typing import List, Optional, Dict, Set, Tuple

class Solution:
    def canMakeSquare(self, grid):
        """
        :type grid: List[List[str]]
        :rtype: bool
        """
        N, W = 3, 2
        return any(max(collections.Counter(grid[i+h][j+w] for h in range(W) for w in range(W)).itervalues()) >= W**2-1
                   for i in range(N-W+1) for j in range(N-W+1))
